### PR TITLE
Do Not Merge: VReplication: instrumentation and logs for debugging data issues 

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
The following functionality has been added to be able to debug data issues found in a recent resharding workflow.

### Sequence number
We add a sequence number to each event sent by vstreamer. On the vreplication side we check that the sequence number is monotonically increasing. This ensures that all vstreamer events reach the apply logic. This validates that events are not dropped either in the network or during some buffering.
Any missing event results in the workflow being stopped and the stream's message updated to an error message with the current event.

### GTID validations
On both the vstreamer and vreplication, we check that gtids are monotonically increasing. If not we log these as warnings: we are not entirely sure if gtids will always be monotonically increasing, hence we don't stop the workflow.

### Check RowsAffected 
For each update and insert we verify the RowsAffected by the applied query. During the copy phase we could have inserts and updates for PKs
which have not already been copied and hence don't need to be applied. For these RowsAffected should be 0, for others non-zero. We log
warnings if the validations failed.

Additionally info logs have been added as part of the apply process. For ease of grepping the logs each log added in this PR contains "dbg:". Errors are logged as edbg:, warnings as wdbg: and info as idbg:

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

## Related Issue(s)

- 

## Checklist
- [ ] Should this PR be backported?
- [ ] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [ ]  VTAdmin

### CODE REVIEW

File: go/mysql/mysql56_gtid_set.go [hunk 1]
```diff
@@ -579,6 +580,36 @@ func (set Mysql56GTIDSet) Difference(other Mysql56GTIDSet) Mysql56GTIDSet {
...
+	for _, intervals := range set {
+		for _, interval := range intervals {
+			txs += interval.end - interval.start + 1
+		}
+	}
+	return txs, nil
+}
+
+func (set Mysql56GTIDSet) Distance(other GTIDSet) (int64, error) {
+	mydbOther, ok := other.(Mysql56GTIDSet)
```
Comment (sougou) [inline go/mysql/mysql56_gtid_set.go:594]: Would it be better to perform a subtract, and then count the num transactions? It will be more accurate.

**Comment by shlomi-noach** [comment]
Since the PR is labeled as "Do Not Merge", I converted it to `Draft`.

File: go/mysql/mysql56_gtid_set.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/mysql/mysql56_gtid_set.go]: Changed to count the transactions reported by the Difference() function.

**Comment by rohit-nayak-ps** [comment]
Closed in favor of https://github.com/vitessio/vitess/pull/7793 which is based off of the problematic cluster's base commit.

**Comment by rohit-nayak-ps** [comment]
Closed in favor of #7793 which is based off of the problematic cluster's base commit.

**Pull request closed without merging**